libvncclient/sockets.c

```diff
@@ -151,10 +153,15 @@ ReadFromRFBServer(rfbClient* client, char *out, unsigned int n)
       if (i <= 0) {
 	if (i < 0) {
 	  if (errno == EWOULDBLOCK || errno == EAGAIN) {
+	    if (++retries > (client->readTimeout * 1000 * 1000 / USECS_WAIT_PER_RETRY))
```

Comment:
This way, there's no way for an application to specifiy "timeout never" by setting `client->readTimeout=0`

Reply:
Good point! I'll provide an update so we can even keep the previous behaviour per default.

Reply:
Should we default `rfbClient::readTimeout` to `0` instead of the proposed `60` seconds?

Reply:
Hmm, maybe yes. Maybe there are users relying on the wait-forever semantics? Dunno. Would the client connect if the network got back?

Reply:
Not sure since I originally wanted to handle cases where a computer is powered off/reset hard and the host's TCP stack never receives any TCP FIN messages or similar. Simply unplugging and replugging a cable shortly after will likely retain TCP connections but even then a timeout could make sense.

Reply:
Mhh, let's keep it safe and try to retain the original behaviour. I.e. timeout 0.